pydantic_ai_slim/pydantic_ai/mcp.py

```diff
@@ -239,6 +239,18 @@ def _map_tool_result_part(
                 )
             else:
                 assert_never(resource)
+        elif isinstance(part, mcp_types.ResourceLink):
+            resource_result: mcp_types.ReadResourceResult = anyio.run(lambda: self._client.read_resource(part.uri))
```

Comment:
Can we make this method `async` so we can `await` here instead of using `anyio.run`?